protob/types.proto

```diff
@@ -270,3 +270,137 @@ message IdentityType {
+}
+
+/**
+ * Structure representing the mosaic attachment for NEM transfer transactions
+ * @used_in NEMTransfer
+ */
+message NEMMosaic {
+	optional string namespace = 1;			// Fully qualified name of the namespace
+	optional string mosaic = 2;			// Name of the mosaic definition
+	optional uint64 quantity = 3;			// Mosaic quantity, always given in smallest units
+}
+
+/**
+ * Structure representing the provision namespace part for NEM transactions
+ * @used_in NEMTransaction
+ */
+message NEMProvisionNamespace {
+	optional string namespace = 1;			// New part concatenated to the parent
+	optional string parent = 2;			// Parent namespace (for child namespaces)
+	optional string sink = 3;			// Rental fee sink address
+	optional uint64 fee = 4;			// Rental fee
+}
+
+/**
+ * Type of levy which will be used for mosaic
+ * @used_in NEMMosaicDefinition
+ */
+enum NEMMosaicLevy {
+	MosaicLevy_Absolute = 1;
+	MosaicLevy_Percentile = 2;
+}
+
+/**
+ * Structure representing the mosaic definition creation part for NEM transactions
+ * @used_in NEMTransaction
+ */
+message NEMMosaicCreation {
+	optional NEMMosaicDefinition definition = 1;	// Mosaic definition
+	optional string sink = 2;			// Creation fee sink address
+	optional uint64 fee = 3;			// Creation fee
+}
+
+/**
+ * Structure representing a mosaic definition
+ * @used_in NEMMosaicCreation
+ */
+message NEMMosaicDefinition {
+	optional string name = 1;			// User-friendly name of the mosaic (for whitelisted mosaics)
+	optional string ticker = 2;			// Ticker of the mosaic (for whitelisted mosaics)
+	optional string namespace = 3;			// Fully qualified name of the namespace
+	optional string mosaic = 4;			// Name of the mosaic definition
+	optional uint32 divisibility = 5;		// Number of decimal places that a mosaic can be divided into
+	optional NEMMosaicLevy levy = 6;		// Levy type
+	optional uint64 fee = 7;			// Levy fee (interpretation depends on levy type)
+	optional string levy_address = 8;		// Levy address
+	optional string levy_namespace = 9;		// Fully qualified name of the namespace of the levy mosaic
+	optional string levy_mosaic = 10;		// Name of the levy mosaic
+	optional uint64 supply = 11;			// Initial supply to create, always given in entire units
+	optional bool mutable_supply = 12;		// Mutable supply
+	optional bool transferable = 13;		// Mosaic allows transfers among accounts other than the creator
+	optional string description = 14;		// Mosaic description
+	repeated uint32 networks = 15;			// Networks that the mosaic is valid on (for whitelisted mosaics)
+}
+
+/**
+ * Structure representing the mosaic supply change part for NEM transactions
+ * @used_in NEMTransaction
+ */
+message NEMMosaicSupplyChange {
+	optional string namespace = 1;			// Fully qualified name of the namespace
+	optional string mosaic = 2;			// Name of the mosaic definition
+	optional NEMSupplyChangeType type = 3;		// Type of supply change
+	optional uint64 delta = 4;			// Supply delta
```

Comment:
Don't you want to use sint64 for delta like you use sint32 below for NEMAggregateModification? (And we can drop the NEMSupplyChangeType). Just asking ...

Reply:
It's simpler this way, because I am keeping it the way it is serialized (`relative_change` is serialized as a signed integer, `delta` is serialized as unsigned with another field for the supply change type)

Reply:
Makes sense.